dags/mag_and_friends.py

```diff
@@ -0,0 +1,491 @@
+                                f"{studyPath}/out/metatdenovo_out",
+                                gzip=False,
+                                task_id = "copy_metatdenovo_results_from_cluster",
+                                task_group=current_tasks
+                            )
+
+                            # this should make rda files for the r package via a dedicated Rscript
+                            post_process_results = BashOperator(
+                                task_id="post_process_results",
+                                bash_command=f"Rscript /data/MicrobiomeDB/mbio_airflow_dags/bin/mag_postProcessing.R {studyName} {studyPath}/out",
+                                task_group=current_tasks
+                            )
+
+                            # this should either update or add a row to the study provenance file
+                            def update_provenance_callable():
+
+                                fieldnames = ['study', 'timestamp', 'mag_revision', 'metatdenovo_revision', 'taxprofiler_revision']
+                                # Update or append new data
+                                with open(PROVENANCE_PATH, 'w', newline='') as csvfile:
+                                    writer = csv.DictWriter(csvfile, fieldnames=fieldnames)
+                                    writer.writeheader()
+                                    
+                                    updated_study = {
+                                        'study': studyName,
+                                        'timestamp': current_timestamp,
+                                        'mag_revision': MAG_VERSION,
+                                        'taxprofiler_revision': TAXPROFILER_VERSION,
+                                        'metatdenovo_revision': METATDENOVO_VERSION
+                                    }
+
+                                    # Merge with existing data
+                                    all_data = {d['study']: d for d in processed_studies}
+                                    all_data.update({studyName: updated_study})
+
+                                    # Write back to the file
+                                    writer.writerows(all_data.values())
+
+                            update_provenance = PythonOperator(
+                                task_id="update_provenance",
+                                python_callable=update_provenance_callable,
+                                task_group=current_tasks
+                            )
+
+                            copy_study_to_cluster >> \
+                            run_mag >> \
+                            watch_mag >> \
+                            update_provenance
+
+                            copy_study_to_cluster >> \
+                            run_metatdenovo >> \
+                            watch_metatdenovo >> \
+                            copy_metatdenovo_results_from_cluster >> \
```

Comment:
I remember you mentioned there is a need to clean up remote directory on pmacs, is that something needed?

Reply:
yea, generally we dont want to leave huge amounts of data on pmacs. theyll charge for it i believe. but also, we need to balance that against compute time. specifically, in order for the `-resume` flag to work for a nextflow run, we need the work dir to still exist on pmacs. so i think while were in a place where were developing and heavily testing this thing, its ok to not clean up remote for now. once were happy w things, we should probably have a user param whether they want to clean up after or not. and we should make an issue for that now, in this repo, so we dont forget about it.

Reply:
I see. I did run into nextflow failed on PMACS once at the QIIME2 steps, 1 out of 100+ test runs. Learning something today!